packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -343,32 +336,38 @@ export class MarketOperationUtils {
             feeSchedule: opts.feeSchedule,
         });
         // Find the optimal path.
-        const optimalPath = findOptimalPath(side, paths, inputAmount, opts.runLimit) || [];
-        // TODO(dorothy-zbornak): Ensure the slippage on the optimal path is <= maxFallbackSlippage
-        // once we decide on a good baseline.
+        let optimalPath = findOptimalPath(side, paths, inputAmount, opts.runLimit) || [];
         if (optimalPath.length === 0) {
             throw new Error(AggregationError.NoOptimalPath);
         }
         // Generate a fallback path if native orders are in the optimal paath.
-        let fallbackPath: Fill[] = [];
         const nativeSubPath = optimalPath.filter(f => f.source === ERC20BridgeSource.Native);
         if (opts.allowFallback && nativeSubPath.length !== 0) {
-            // The fallback path is, at most, as large as the native path.
-            const fallbackInputAmount = BigNumber.min(inputAmount, getPathSize(nativeSubPath, inputAmount)[0]);
-            fallbackPath =
-                findOptimalPath(side, getFallbackSourcePaths(optimalPath, paths), fallbackInputAmount, opts.runLimit) ||
-                [];
+            const nonNativePaths = paths.filter(p => p.length > 0 && p[0].source !== ERC20BridgeSource.Native);
+            // The fallback path is the entire input amount
+            const nonNativeOptimalPath = findOptimalPath(side, nonNativePaths, inputAmount, opts.runLimit) || [];
             const fallbackSlippage = getPathAdjustedSlippage(
                 side,
-                fallbackPath,
-                fallbackInputAmount,
+                nonNativeOptimalPath,
+                inputAmount,
                 getPathAdjustedRate(side, optimalPath, inputAmount),
             );
-            if (fallbackSlippage > maxFallbackSlippage) {
-                fallbackPath = [];
+            if (fallbackSlippage <= maxFallbackSlippage) {
+                // If the last fill is Native and penultimate is not
+                // then this intention is a partial fill
+                // In this case we drop it as we don't want it to fail
+                // at the end and we don't want to fully fill it
+                // tslint:disable-next-line: no-unused-variable
+                const [last, penultimate, ..._rest] = optimalPath.reverse();
```

Comment:
We can just drop the `_rest` var, yeah?

Reply:
Even though we don't use the value of `optimalPath` again in the code, might be a god idea to do `optimalPath.slice().reverse()` since `reverse()` is a mutator and might trip up other people expecting `optimalPath` to still be the same at this point.